lib/actions/options/read-only.ts

```diff
@@ -22,8 +30,52 @@ export interface OptionsReadOnlyActions {
 export class ContractsOptionsReadOnlyActions implements OptionsReadOnlyActions {
   constructor(private roContracts: ReadOnlyContracts) {}
 
+  async fetchTokenNames(
```

Comment:
Currently, we use the same name for each option/obligation. We could give it a name based on expiryTime and underlying etc. though.
See code: https://github.com/interlay/xopts/blob/dev/contracts/Option.sol#L62